extension/tpcds/tpcds-extension.cpp

```diff
@@ -0,0 +1,198 @@
+#include "duckdb/parser/parser.hpp"
+#include "duckdb/parser/statement/select_statement.hpp"
+#endif
+
+namespace duckdb {
+
+struct DSDGenFunctionData : public TableFunctionData {
+	DSDGenFunctionData() {
+	}
+
+	bool finished = false;
+	int sf = 0;
+	string schema = DEFAULT_SCHEMA;
+	string suffix;
+	bool overwrite = false;
+};
+
+static unique_ptr<FunctionData> DsdgenBind(ClientContext &context, vector<Value> &inputs,
+                                           unordered_map<string, Value> &named_parameters,
+                                           vector<LogicalType> &input_table_types, vector<string> &input_table_names,
+                                           vector<LogicalType> &return_types, vector<string> &names) {
+	auto result = make_unique<DSDGenFunctionData>();
+	for (auto &kv : named_parameters) {
+		if (kv.first == "sf") {
+			result->sf = kv.second.value_.integer;
+		} else if (kv.first == "schema") {
+			result->schema = kv.second.str_value;
+		} else if (kv.first == "suffix") {
+			result->suffix = kv.second.str_value;
+		} else if (kv.first == "overwrite") {
+			result->overwrite = kv.second.value_.boolean;
+		}
+	}
+	return_types.push_back(LogicalType::BOOLEAN);
+	names.emplace_back("Success");
+	return move(result);
+}
+
+static void DsdgenFunction(ClientContext &context, const FunctionData *bind_data, FunctionOperatorData *operator_state,
+                           DataChunk *input, DataChunk &output) {
+	auto &data = (DSDGenFunctionData &)*bind_data;
+	if (data.finished) {
+		return;
+	}
+	tpcds::DSDGenWrapper::DSDGen(data.sf, context, data.schema, data.suffix);
+
+	data.finished = true;
+}
+
+struct TPCDSData : public FunctionOperatorData {
+	TPCDSData() : offset(0) {
+	}
+	idx_t offset;
+};
+
+unique_ptr<FunctionOperatorData> TPCDSInit(ClientContext &context, const FunctionData *bind_data,
+                                           const vector<column_t> &column_ids, TableFilterCollection *filters) {
+	auto result = make_unique<TPCDSData>();
+	return move(result);
+}
+
+static unique_ptr<FunctionData> TPCDSQueryBind(ClientContext &context, vector<Value> &inputs,
+                                               unordered_map<string, Value> &named_parameters,
+                                               vector<LogicalType> &input_table_types,
+                                               vector<string> &input_table_names, vector<LogicalType> &return_types,
+                                               vector<string> &names) {
+	names.emplace_back("query_nr");
+	return_types.push_back(LogicalType::INTEGER);
+
+	names.emplace_back("query");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	return nullptr;
+}
+
+static void TPCDSQueryFunction(ClientContext &context, const FunctionData *bind_data,
+                               FunctionOperatorData *operator_state, DataChunk *input, DataChunk &output) {
+	auto &data = (TPCDSData &)*operator_state;
+	idx_t tpcds_queries = 22;
```

Comment:
tpcds has more queries than 22 no?

Reply:
Oops, copy and pasted a bit too much from the TPC-H extension 😉